### PR TITLE
Expose BlockLocator and Libplanet.Net related functions in BlockChain<T>

### PR DESCRIPTION
**Pull request by tkiapril**
(Partially?) Implements #1762.
This PR does not completely decouple Libplanet.Net from Libplanet, but it exposes few APIs required to remove `[assembly: InternalsVisibleTo("Libplanet.Net")]` to the public. The exposed APIs are those of which does not seem to introduce severe side effects.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -16,6 +16,13 @@ To be released.
 
 ### Added APIs
 
+ - Added `BlockLocator` class. [[#1762]]
```
Comment (greymistcube) [inline CHANGES.md:19]:
```suggestion
 - Added `BlockLocator` class. [[#1762], [#2140]]
```

The usual format is `\[[\[#<issue_number>\], ]\[#<pr_number>\]\]`. 😗

**Comment by greymistcube** [comment]
Seems almost set. Just needs rebasing. 🙂

**Review by greymistcube — APPROVED** [review]

File: CHANGES.md [hunk 2]
```diff
@@ -16,6 +16,13 @@ To be released.
 
 ### Added APIs
 
+ - Added `BlockLocator` class. [[#1762, #2140]]
```
Comment (longfin) [inline CHANGES.md:19]:
```suggestion
 - Added `BlockLocator` class. [[#1762], [#2140]]
```

File: CHANGES.md [hunk 3]
```diff
@@ -16,6 +16,13 @@ To be released.
...
+ - Added methods required for decoupling `Libplanet.Net` from
+   `Libplanet`. [[#1762, #2140]]
```
Comment (longfin) [inline CHANGES.md:21]:
```suggestion
   `Libplanet`. [[#1762], [#2140]]
```

**Review by longfin — APPROVED** [review]

**Review by riemannulus — APPROVED** [review]

**Pull request merged**